src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -283,6 +284,63 @@ public class LockFileBuilder
             return lockFile;
         }
 
+        private void ValidateLockFileLibrary(IList<LockFileLibrary> libraries)
+        {
+            ILookup<Tuple<string, Versioning.NuGetVersion>, LockFileLibrary> librariesByNameAndVersion = libraries.ToLookup(lib => Tuple.Create(lib.Name, lib.Version));
```

Comment:
Cleanup the types. 

Refer to rule #26 - https://github.com/NuGet/NuGet.Client/blob/dev/docs/coding-guidelines.md

Reply:
Given that this is a hot path and an unlikely occurrence at that, any way we can make it cheaper allocation wise?

Reply:
@nkolev92 
Please check current implementation. If you think we need some kind of actual `perf` data then please let me know. I was working on it but something else need attention.